src/integrations/pulsar/consumer.cpp

```diff
@@ -0,0 +1,285 @@
+// Copyright 2021 Memgraph Ltd.
```

Comment:
This file is really similar to the Kafka consumer... What is the biggest argument against extracting the common logic?

Reply:
Mostly different error handling and message committing which is scattered around the more complex code like `StartConsuming` and `Check`. For some solutions, it was worse than this so I think that we would just overcomplicate it more with extracting common logic.